text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+similar to a credit card validation. It's implemented as an extensible function 
+such that further validation steps can be added, if needed. For details on the 
+equation see: [Check digit validation](https://www.gs1us.org/tools/check-digit-calculator "Check digit validation")
+
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+[state]: #state
+
+### Location Representation
+
+The primary object stored in state is �Location�, which consists of a 
+location_id (GS1 GLN), an owner (org_id compatible with Pike), and a list of 
+property name/value pairs. The properties available are defined by the Grid 
+Location Schema Transaction Family and are restricted to the fields and rules of
+the GS1 Location Property Schema. Transactions which are responsible for setting
+location state values must ensure that the properties conform with the 
+requirements of the GS1 Location Property Schema.
+
+The GS1 GLN is an identifier (location_id) used to identify entities or 
+locations. A GLN is the data transmitted from a data carrier (barcode, rfid, 
+etc.) scan and is made up of a company prefix, location reference, and check 
+digit.
+
+### Referencing Locations
+
+Locations are uniquely referenced by their location_id and location_namespace.
+For GS1, locations are referenced by the GLN identifier.
+
+**Note: For the remainder of this document, �address� will refer to a 
+Grid/Transact address as opposed to a Street Address.**
+
+### Location Addressing in the Merkle-Radix State System
+
+In order to uniquely locate GS1 locations in the Merkle-Radix state system, an 
+address must be constructed which identifies the storage location of the 
+Location representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix �621dee�.
+Locations are further prefixed under the Grid namespace with reserved 
+enumerations of �04� (�00�, �01�, �02� and �03� being reserved for other 
+purposes) indicating �Locations� and an additional �01� indicating �GS1 
+Locations�.
+
+Therefore, all addresses starting with �621dee� + �04� are Grid locations, and 
+more specifically, all addresses starting with �621dee� + �04� + �01� are Grid 
+GS1 Locations identified by a GLN.
+
+The GLN format consists of a 13-digit �numeric string� that follows a similar 
+structure to GTIN-13. The GLN structure first includes a GS1 Company Prefix 
+which is 7-10 digits in length and is assigned by a GS1 Member Organization. A 
+Location Reference number follows which is 2-5 digits in length and is 
+allocated by the company to a location or party. Lastly, a single digit Check 
+Digit is calculated and applied according to a GS1 algorithm.
+
+![GLN Data Format](https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSFo3J5QULgkCWyIH3yeddRPx1H5aQusLRRq47y1n2FqTshGxs%3Ax-raw-image%3A%2F%2F%2Ff4fc819afc288cee1948ca0d40d76c3bcd56a6d270fcfeeac80fd25784eb6071&usqp=CAU)
```

Comment:
Instead of showing this inline, it would be better to make it clear that it is not part of this document from a copyright perspective and make it a normal link you have to click to see it.